### PR TITLE
Specify an optional list of MAC addresses to assign on new VMWare instances

### PR DESCRIPTION
**Pull request by vinseon**
Informations:
 - Modifications triggered by the new MAC address feature from  ow2-proactive/connector-iaas#26
 - The new field is currently assigned to VMWare infrastructure only and is optional.
 - Multiple MAC addresses can be specified using ',' separator.

### CODE REVIEW

File: infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java [hunk 1]
```diff
@@ -217,12 +217,13 @@ public void acquireNode() {
 
         Set<String> instancesIds = Sets.newHashSet();
 
+        // MAC addresses not yet managed
```
Comment (marcocast) [inline infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java]: Do we need this comment?

File: infrastructures/infrastructure-vmware/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/VMWareInfrastructure.java [hunk 2]
```diff
@@ -99,6 +99,9 @@
     @Configurable(description = "Command used to download the worker jar")
     protected String downloadCommand = generateDefaultDownloadCommand();
 
+    @Configurable(description = "Optional list of MAC addresses separated by comma ',' to assign on new cloned VMs (1 per instance)")
```
Comment (marcocast) [inline infrastructures/infrastructure-vmware/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/VMWareInfrastructure.java]: should we mention here that the number of mac addresses need to match with the number of instances?

File: infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java (hunk 1 above)
Comment (vinseon) [inline infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java]: Removed in https://github.com/vinseon/nodesource-addons/commit/ac0f9823268618d9e9dd5b1a20c4559b94364591

File: infrastructures/infrastructure-vmware/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/VMWareInfrastructure.java (hunk 2 above)
Comment (vinseon) [inline infrastructures/infrastructure-vmware/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/VMWareInfrastructure.java]: Removed in https://github.com/vinseon/nodesource-addons/commit/ac0f9823268618d9e9dd5b1a20c4559b94364591

**Comment by vinseon** [comment]
@marcocast both implementations use the same method createInstancesWithOptions from ConnectorIaasController, we thus only need to specify a null MAC address.
I think its fine to keep it like this to ease future implementation in openstack.
Although I agree it will not scale with plenty of infrastructures dedicated options, its the only extra option for now.

**Comment by mbenguig** [comment]
Ok

**Comment by mbenguig** [comment]
Ok

**Pull request merged**